st2api/st2api/controllers/v1/actions.py

```diff
@@ -94,9 +95,16 @@ def post(self, action):
             Handles requests:
                 POST /actions/
         """
-        # Perform validation
-        validate_not_part_of_system_pack(action)
-        action_validator.validate_action(action)
+
+        try:
+            # Perform validation
+            validate_not_part_of_system_pack(action)
+            action_validator.validate_action(action)
+        except (ValidationError, ValueError,
+                ValueValidationException, InvalidActionParameterException) as e:
+            LOG.exception('Unable to create action data=%s', action)
+            abort(http_client.BAD_REQUEST, str(e))
+            return
```

Comment:
If I don't do this, then API will come back with 500 instead of bad request with the actual fault message.